pkg/storage/buckets/store.go

```diff
@@ -98,28 +99,43 @@ func (b *BucketStore) Get(ctx context.Context, bucket string) (meta Meta, err er
 	return convertMeta(objMeta)
 }
 
-// Put calls objects store Put
-func (b *BucketStore) Put(ctx context.Context, bucket string, pathCipher storj.Cipher) (meta Meta, err error) {
+// Put calls objects store Put and fills in some specific metadata to be used
+// in the bucket's object Pointer. Note that the Meta.Created field is ignored.
+func (b *BucketStore) Put(ctx context.Context, bucketName string, inMeta Meta) (meta Meta, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	if bucket == "" {
+	if bucketName == "" {
 		return Meta{}, storj.ErrNoBucket.New("")
 	}
 
+	pathCipher := inMeta.PathEncryptionType
 	if pathCipher < storj.Unencrypted || pathCipher > storj.SecretBox {
 		return Meta{}, encryption.ErrInvalidConfig.New("encryption type %d is not supported", pathCipher)
 	}
 
 	r := bytes.NewReader(nil)
 	userMeta := map[string]string{
-		"path-enc-type": strconv.Itoa(int(pathCipher)),
+		"path-enc-type":     strconv.Itoa(int(pathCipher)),
+		"default-seg-size":  strconv.FormatInt(inMeta.SegmentsSize, 10),
+		"default-enc-type":  strconv.Itoa(int(inMeta.EncryptionScheme.Cipher)),
+		"default-enc-blksz": strconv.FormatInt(int64(inMeta.EncryptionScheme.BlockSize), 10),
+		"default-rs-algo":   strconv.Itoa(int(inMeta.RedundancyScheme.Algorithm)),
+		"default-rs-sharsz": strconv.FormatInt(int64(inMeta.RedundancyScheme.ShareSize), 10),
+		"default-rs-reqd":   strconv.Itoa(int(inMeta.RedundancyScheme.RequiredShares)),
+		"default-rs-repair": strconv.Itoa(int(inMeta.RedundancyScheme.RepairShares)),
+		"default-rs-optim":  strconv.Itoa(int(inMeta.RedundancyScheme.OptimalShares)),
+		"default-rs-total":  strconv.Itoa(int(inMeta.RedundancyScheme.TotalShares)),
```

Comment:
I'm not really liking the shorthands and strings for the keys here. I'm wondering whether we should a have a separate protobuf for bucket information that can be serialized?

Reply:
This would mean we need to write a migration for that later.

Reply:
We would anyway need to think about this when finish the metainfo refactoring.